docs/fn-2.0/README.md

```diff
@@ -0,0 +1,416 @@
+passes them off to trigger manager / LB / runner.
+
+#### Scheduler
+
+On schedule set by user, the trigger is fired.
+Scheduler would have its own API to manage schedule -> trigger mappings.
+
+#### Queue
+
+Listens on a stream/topic/channel, can have a filter
+For every message that passes filter, fire the trigger
+
+## Example flow
+
+HTTP
+
+### HTTP Router receives request
+
+Request comes into HTTP router under pretty url, e.g. `http://example.com/repomanager`
+Convert request to CloudEvent, eg:
+
+```json
+{
+  "cloud-events-version": "0.1",
+  "event-id": "6480da1a-5028-4301-acc3-fbae628207b3",
+  "source": "http://example.com/repomanager",
+  "event-type": "http-request",
+  "event-type-version": "v1.5",
+  "event-time": "2018-04-01T23:12:34Z",
+  "content-type": "application/json",
+  "data": {
+     REQUEST_BODY
+  },
+  “extensions”: {
+    “protocol”: {
+       “headers”: …
+       Etc, same as we do now for JSON format. 
+    }
+  }
+}
+```
+
+Pass event to trigger manager.
+
+### Trigger manager (TM) receives event
+
+TM looks up the matching trigger.
+Trigger has function URI + config data. 
+TM also get app config data.
+Merges config and event to get this event:
+
+```json
+{
+  "cloud-events-version": "0.1",
+  "event-id": "6480da1a-5028-4301-acc3-fbae628207b3",
+  "source": "http://example.com/repomanager",
+  "event-type": "http-request",
+  "event-type-version": "v1.5",
+  "event-time": "2018-04-01T23:12:34Z",
+  "content-type": "application/json",
+  "data": {
+     REQUEST_BODY
+  },
+  “extensions”: {
+    “protocol”: {
+       “headers”: …
+       Etc, same as we do now for JSON format. 
+    },
+    “config”: {
+        ALL APP AND TRIGGER CONFIGS
+     }
+     “function”: “https://funcs.fnproject.io/NAMESPACE/FUNC:VERSION”
+  }
+}
+```
+
+Passes event to function router (aka LB) OR runner (whichever the trigger manager is configured for, they both have the same /run endpoint. Single machine setup would be direct to runner). 
+
+### Function Router gets event
+
+Picks runner to run it on and passes along event
+
+### Runner gets event
+
+Takes function URL and looks up metadata from function registry. 
+Executes the function
+Returns CloudEvent with response as “data” body, strips out all the extension stuff and replace with response specific info.
+
+```json
+{
+  "cloud-events-version": "0.1",
+  "event-id": "6480da1a-5028-4301-acc3-fbae628207b3",
+  "source": "http://example.com/repomanager",
+  "event-type": "http-request",
+  "event-type-version": "v1.5",
+  "event-time": "2018-04-01T23:12:34Z",
+  "content-type": "application/json",
+  "data": {
+     RESPONSE_BODY
+  },
+  “extensions”: {
+    “protocol”: {
+       “headers”: …
+       Etc, same as we do now for JSON format responses 
+    },
+  }
+}
+```
+
+## CLI changes
+
+### fn deploy
+
+func.yaml barely changes:
+
+```yaml
+function:
+  name: yodawg
+  image: hub.docker.com/fnproject/hello
+  version: 0.0.7
+  memory: 42
+
+trigger:
```

Comment:
shouldn't `triggers` be a list of actual triggers?
```yaml
triggers:
 - name: myApp/sayhello
   type: http
 - name: myApp/sayhello
   type: queue
```

Reply:
yea, good idea.